src/compiler/checker.ts

```diff
@@ -177,7 +177,8 @@ namespace ts {
         SkipGenericFunctions = 1 << 3,                  // Skip single signature generic functions
         IsForSignatureHelp = 1 << 4,                    // Call resolution for purposes of signature help
         IsForStringLiteralArgumentCompletions = 1 << 5, // Do not infer from the argument currently being typed
-        RestBindingElement = 1 << 6,                    // Checking a type that is going to be used to determine the type of a rest binding element
+        IsForObjectLiteralArgumentCompletions = 1 << 6, // Use the constraint instead of the default type for object literal argument completions

```

Comment:
I imagine that the same thing happens for array literals - could we cover for them in this PR?

Reply:
I've renamed the flag